File: frontend/nomia-abslout-POS/src/queries.js
import gql from 'graphql-tag'

export const SITE_INFO = gql`
  query {
    site {
      name
    }
  }
`

export const All_SURVEYS = gql`
  query{
    all_surveys {
      id_sur
      question_text
      description
      active
      slug
      choice {
        id_choice
        choice_text
        slug
      }
    }
  }
`

export const All_QUESTIONS = gql`
  query{
    all_questions {
      id_ques
      question_text
      description
      active
      slug
      choice {
        id_choice
        choice_text
        slug
      }
    }
  }
`

export const SURVEY_BY_SLUG = gql`
  query($slug: String!){
    survey_by_slug(slug: $slug) {
      id_sur
      question_text
      description
      active
      slug
    }
  }
`

export const QUESTION_BY_SLUG = gql`
  query($slug: String!) {
    question_by_slug(slug: $slug) {
      id_ques
      question_text
      description
      active
      slug
    }
  }
`

export const CHOICE_SURVEY_BY_SLUG = gql`
  query($slug: String!) {
    choice_survey_by_slug(slug: $slug) {
      id_choice
      survey
      choice_text
      slug
    }
  }
`

export const CHOICE_QUESTION_BY_SLUG = gql`
  query($slug: String!) {
    choice_question_by_slug(slug: $slug) {
      id_choice
      question
      choice_text
      slug
    }
  }
`

export const CURRENT_USER = gql`
  query ($username: String!) {
    currentUser(username: $username) {
      id
      username
      email
      answer_survey {
        choices
      }
      answer_question {
        choices
      }
    }
  }
`;

File: backend/onboarding/queries.py
import graphene
from . import models
from . import types


class Query(graphene.ObjectType):
    site = graphene.Field(types.SiteType)
    all_user = graphene.List(types.UserType)
    current_user = graphene.Field(types.UserType, username=graphene.String())

    all_surveys = graphene.List(types.SurveyType)
    all_questions = graphene.List(types.QuestionType)

    survey_by_slug = graphene.List(types.SurveyType, slug=graphene.String())
    question_by_slug = graphene.List(types.QuestionType, slug=graphene.String())

    choice_survey_by_slug = graphene.List(types.ChoiceSurveyType, slug=graphene.String())
    choice_question_by_slug = graphene.List(types.ChoiceQuestionType, slug=graphene.String())

    def resolve_site(root, info):
        return (
            models.Site.objects.first()
        )

    def resolve_all_user(root, info):
        return (
            models.User.objects.all()
        )

    def resolve_current_user(root, info, username):
        return (
            models.User.objects.get(username__iexact=username)
        )

    def resolve_all_surveys(root, info):
        return (
            models.Survey.objects.all()
        )

    def resolve_all_questions(root, info):
        return (
            models.Question.objects.all()
        )

    def resolve_survey_by_slug(root, info, slug):
        return (
            models.Survey.objects.get(slug__iexact=slug)
        )

    def resolve_question_by_slug(root, info, slug):
        return (
            models.Question.objects.get(slug__iexact=slug)
        )

    def resolve_choice_survey_by_slug(root, info, slug):
        return (
            models.ChoiceSurvey.objects.get(slug__iexact=slug)
        )

    def resolve_choice_question_by_slug(root, info, slug):
        return (
            models.ChoiceQuestion.objects.get(slug__iexact=slug)
        )